### ISSUE TITLE
Managed experiment GUI has no elapsed run time.

### ISSUE THREAD

**Issue by rfoxkendo** [FRIBDAQ/NSCLDAQ#101]
Simple solutions to this fail in the presence of multiple GUIs and GUIS starting in the middle of an experiment.   The preferred fix to this is a bit big to put into 12.0 so here's my proposal:

### 12.0 interim fix:

  The configuration database has a table for  'user' variables.  That's a key value store.  Oddly enough it's named ```kvstore`` This can be used in an interim way that does not affect much:
*   Add a program called e.g. ```mg_stamper```  It sets a user variable, let's call it ```last_run_start``` to the current time.
*   Users who want elapsed run times in their GUIs, add ```mg_stamper``` to the end of their ````BEGIN``` sequence so, if not null, it contains the start time of the most recent transition to BEGIN (run is active).
*   Modify the Run control GUI to attempt to fetch the value of this user variable when the state is detected either to have change to ```BEGIN``` or when the program is started and the state is already ```BEGIN``` use the time in this variable as the start of run time and provide a running elapsed time relative to that.  If ```last_run_start``` is either not defined or it is unreasonable (e.g. 0), then just display that the elapsed run time  is not available.

### 12.1 fix:

Add a table managed by the manager server call it e.g. ```state_transition_log``` it's schema might be:

```sql
CREATE TABLE IF NOT EXISTS state_transition_log (
  id INTEGER PRIMARY KEY
  to_state INTEGER
  time INTEGER
);
```

Where:
*  id is just the primary key tables like to have.
* to_state is a foreign key into the state name table identifying the state being transitioned into.
* time is e.g. the **clock seconds** at which the transition completed. 

This table can be maintained by the state transition part of the system.  The start time of the most recent run can be gotten with the following bit of SQL Query magic:

```sql
SELECT time FROM state_transition_log
INNER JOIN transition_name ON state_transition_log.to_state = transition_name.id
WHERE transition_name.name = 'BEGIN'
ORDER by state_transition_log.id DESC
LIMIT 1
```
 
I suppose, techincally one should not assume ids are monotonically assigned  and the ORDER clause could be on state_transition_log.time which *is* going to be monotonic.

This bit of SQL magic finds the time stamp of the most recent transition to BEGIN.

This could be queried in the same way/time to establish the start run time for GUIs as in the 12.0 interim fix.

The additional benefit of this is that one can generate a log for he users of when runs started and stopped.  One could add fields to the state_transition_log table to hold the run number and title of the run (those get stored in the ```kvstore``` table) by the BEGIN sequence.

Note that the managed system does not support (out of the box) a PAUSE state (though very knowledgable users *could* theoretically add that.  PAUSE/RESUME, of course complicate this 'simple' scheme quite a bit.

**Comment by rfoxkendo** [comment]
Starting on the 12.0 part of this issue.  Will be done on
rfoxkenod/issue101/120

**Comment by rfoxkendo** [comment]
I've got the 12.0 part fixed.  I'm going to do a pull request on rfoxkendo/issue101/120 -> 12.0 maint.  I think the code used by the client just needs to be tuned a bit (modify the ElapsedTime module).

**Comment by rfoxkendo** [comment]
Solved for NSCLDAQ 12.1 as described in the proposal with the added field in transition_log of status which is 1 on a successful trasition and 0 on a failed transition.
